File: backend/src/main/java/com/app/notes/model/Note.java
package com.app.notes.model;

import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.Date;

@Data
public class Note {

    @Id
    int id_note;
    String title;
    String content;
    Date lastUpdated;
    boolean archived;
    boolean deleted;

    public boolean getArchived(){
        return this.archived;
    }
    public boolean getDeleted(){
        return this.deleted;
    }

}

File: frontend/notes/src/app/models/note.ts
export class Note {
    id_note: number = 0;
    title:string;
    content:string;
    lastUpdated:Date;
    archived:boolean;
    deleted:boolean;

    constructor(t:string,c:string, l:Date, archived:boolean, d:boolean){
        this.title=t;
        this.lastUpdated=l;
        this.content=c;
        this.archived=archived;
        this.deleted=d;
    }
}
